### PR TITLE
Review curve25519 (DO NOT MERGE !)

### PR DESCRIPTION
**Pull request by beurdouche**
DO NOT MERGE !

### CODE REVIEW

**Comment by beurdouche** [comment]
Per request from Ekr, multiple rounds of reviews on Curve25519 starting by a coding style...
Inlined suggestions in a PR as we don't have a proper code review tool ! We need one !

**Comment by jkzinzindohoue** [comment]
How important is it to review the Low* code (@prosecco @beurdouche) ? The specs has received a lot of attention, so has the generated C code, and this is not a good time for me to rewrite the Low* code of curve25519 (although I admit that from a style standpoint it could be improved).
Also I am not in favor of the (**) notation as a long term solution. It is convenient at the moment for readability but there are thousands of lines of code, it is just too much to edit.
Instead I wish we could collapse the lines where the expressions return `unit` using the emacs fstar mode.
I will open an issue for that.

**Comment by beurdouche** [comment]
You should not worry about it, there is absolutely no rush. The idea is more to make it a bit cleaner so that people looking at it will not judge the quality of the NSS code by the look of the HaCl code. I can do small coding style improvements by myself without touching the proofs anyway...
I am ok with not adding (**) and to find a better solution, but reading this code is currently very difficult. Also selecting expressions returning unit is not a solution as these could be computationally relevant if they are stateful.

**Comment by jkzinzindohoue** [comment]
Sure, I meant `Tot unit` and `Ghost 'a pre post`

File: specs/Spec.Curve25519.fst [hunk 1]
```diff
@@ -21,8 +22,11 @@ let zero : elem = 0
 let one  : elem = 1
 let ( +@ ) = fadd
 let ( *@ ) = fmul
+// BB: Are the operator definitions used ? Else remove.
+// BB: Add a blank line
 (** Exponentiation *)
 let rec ( ** ) (e:elem) (n:pos) : Tot elem (decreases n) =
+// BB: Is this an operator or a comment ?
```
Comment (beurdouche) [inline specs/Spec.Curve25519.fst]: Ohh that's the name :) Didn't look properly on first pass !

**Pull request closed without merging**